bokehjs/src/lib/models/tools/inspectors/customjs_hover.ts

```diff
@@ -0,0 +1,51 @@
+import {Model} from "../../../model"
+import * as p from "core/properties"
+import {keys, values} from "core/util/object"
+
+declare var exports: {[key: string]: any}
+
+export namespace CustomJSHover {
+  export interface Attrs extends Model.Attrs {
+    args: {[key: string]: any}
+    formatter: string
```

Comment:
It is hover specific, it get's all the "special vars" which is necessary to actually be able to do anything seriously interesting, i.e. aggregate over multiple simultaneous hovers, convert to lat/lon, etc. as well as getting any (optional) format.  It occurred to me that it might be nice if this could be general, i.e. used for axis ticks, and tooltips, and other things, but I just don't think that is the case, so that's why I did not go with `CustomJSFormatter`. I suppose `CustomJSHoverFormatter` is a possibility but it is extremely long-winded.

Reply:
It would be good to do this, if we decide to somehow bring all custom js models under a common API in future.

Reply:
OK I will push this change then plan to merge unless there are other comments